plugin-lifecycle/src/test/java/com/mapbox/maps/extension/lifecycle/MapboxLifecyclePluginImplTest.kt

```diff
@@ -37,4 +42,82 @@ class MapboxLifecyclePluginImplTest {
 
     verify { mapboxLifecycleObserver.onLowMemory() }
   }
+}
+
+@RunWith(ParameterizedRobolectricTestRunner::class)
+@Config(shadows = [ShadowLogger::class])
+class AcceptedTrimMemoryLevelTest(private val level: Int) {
+  private lateinit var mapboxLifecyclePlugin: MapboxLifecyclePlugin
+  private val mapView: FrameLayout = mockk(relaxed = true)
+  private val context: Context = mockk(relaxed = true)
+  private val lifecycleOwner: LifecycleOwner = mockk(relaxed = true)
+  private val slot = slot<ComponentCallbacks2>()
+
+  @Before
+  fun setUp() {
+    mockkStatic(ViewTreeLifecycleOwner::class)
+    every { mapView.context } returns context
+    every { ViewTreeLifecycleOwner.get(any()) } returns lifecycleOwner
+    mapboxLifecyclePlugin = MapboxLifecyclePluginImpl()
+  }
+
+  @Test
+  fun checkOnLowMemoryIsCalled() {
+    val mapboxLifecycleObserver: MapboxLifecycleObserver = mockk(relaxed = true)
+    every { context.registerComponentCallbacks(capture(slot)) } answers {
+      slot.captured.onTrimMemory(level)
+    }
+    mapboxLifecyclePlugin.registerLifecycleObserver(mapView, mapboxLifecycleObserver)
+
+    verify { mapboxLifecycleObserver.onLowMemory() }
+  }
+
+  companion object {
+    @JvmStatic
+    @ParameterizedRobolectricTestRunner.Parameters(name = "TrimMemoryLevel {0} should trigger onLowMemory")
+    fun data() = listOf(
+      ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL,
```

Comment:
what do you think about potentially missed items for this enum?
maybe use reflection?